bigc/api_client.py

```diff
@@ -4,56 +4,88 @@
 
 import requests
 
-from bigc.exceptions import BigCommerceException, BigCommerceNetworkError, GatewayTimeoutError
+from bigc.exceptions import (
+    BadGatewayError,
+    BigCommerceException,
+    BigCommerceNetworkError,
+    GatewayTimeoutError,
+    InternalServerError,
+    ServiceUnavailableError,
+)
 
 MAX_V2_PAGE_SIZE = 250
 MAX_V3_PAGE_SIZE = 250
 
 
 class BigCommerceRequestClient(ABC):
-    def __init__(self, store_hash: str, access_token: str, timeout: float | None = None):
+    def __init__(self, store_hash: str, access_token: str, timeout: float | None = None, retries: int | None = None):
         self.store_hash = store_hash
         self.access_token = access_token
         self.timeout = timeout
+        self.retries = retries
 
     def request(
-        self,
-        method: str,
-        path: str,
-        *,
-        data: Any = None,
-        params: dict[str, Any] | None = None,
-        headers: dict[str, str] | None = None,
-        timeout: float | None = None,
+            self,
+            method: str,
+            path: str,
+            *,
+            data: Any = None,
+            params: dict[str, Any] | None = None,
+            headers: dict[str, str] | None = None,
+            timeout: float | None = None,
+            retries: int | None = None,
     ) -> Any:
-        """Make a request to the BigCommerce API (uses Requests internally)"""
+        method = method.upper()
 
         if headers is None:
             headers = {}
         if timeout is None:
             timeout = self.timeout
+        if retries is None and method != 'POST':
+            retries = self.retries if self.retries is not None else (2 if method == 'GET' else 0)
```

Comment:
I'm concerned that if you `.post(..., retries=3)` then it will silently not retry, rather than alerting the caller that this is an invalid signature.

Reply:
<img width="696" alt="image" src="https://github.com/user-attachments/assets/d78677bf-ce07-46ed-9872-5cd74f23045a" />